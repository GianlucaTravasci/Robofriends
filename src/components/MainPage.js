import React, { Component } from 'react';
import Header from '../components/Header';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';

export class MainPage extends Component {
    componentDidMount() {
        this.props.onRequestRobots();
    }
    filteredRobots = () => {
        const {  robots, searchField } = this.props;
        robots.filter(robot => {
            return robot.name.toLowerCase().includes(searchField.toLowerCase());
        })
    }
    render() {
        const { onSearchChange, isPending } = this.props;
        
        return (
          <div className='tc'>
            <Header />
            <SearchBox searchChange={onSearchChange}/>
            <Scroll>
              { isPending ? <h1>Loading</h1> :
                <ErrorBoundry>
                  <CardList robots={filteredRobots} />
                </ErrorBoundry>
              }
            </Scroll>
          </div>
        );
    }
}

export default MainPage;