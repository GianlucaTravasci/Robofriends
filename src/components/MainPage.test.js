import { shallow } from 'enzyme';
import React from 'react';

import MainPage from './MainPage';

let wrapper;

beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchField: '',
        isPending: false,
        error: ''
    }
    wrapper = shallow(<MainPage {...mockProps} />);
})

it('render main page without crashing', () => {
    expect(wrapper.debug()).toMatchSnapshot();
}) 
