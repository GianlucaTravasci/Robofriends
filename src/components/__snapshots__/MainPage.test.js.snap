// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render main page without crashing 1`] = `
"<div className=\\"tc\\">
  <Header />
  <SearchBox searchChange={[undefined]} />
  <Scroll>
    <ErrorBoundry>
      <CardList robots={{...}} />
    </ErrorBoundry>
  </Scroll>
</div>"
`;
